### PR TITLE
Create onboarding bootcamp

### LINKED ISSUE #1997: Create onboarding tasks/boot camp materials for new developers
### Please confirm that you have searched existing issues in the repo

Yes, I have searched the existing issues

### Any related issues?

_No response_

### What is the area that this feature belongs to?

Documentation

### Is your feature request related to a problem? Please describe.

New developers are often confused about how to get started in brownfield projects. Same for MarkBind, which has multiple packages and a number of ways to make contributions to the codebase.

### Describe the solution you'd like

Similar to how companies have onboarding tasks/bootcamp to ramp up developer familiarity with the codebase, let's create a series of tasks with checkpoints to guide new developers from setting up to developing a small sample feature. This aims to show developers a standard development workflow, as well as some considerations when navigating our codebase.

Some ideas:
- Objective 1: Setup and explore MarkBind as a user
  - Task a: create a MarkBind site, specify some MarkBind components/syntax to be used in some content, and deploy it to GitHub pages
- Objective 2: Understand how to contribute to our documentation
  - Task a: make an edit to our UG/DG, run our docs locally to verify the changes
- Objective 3: Add tests to functional test cases
  - Task a: add a new test file, update the test content, and verify that the test cases have been updated correctly
- Objective 4: Implement a new feature
  - Task a: add a new text-style syntax addition
  - Task b: add a new vue component

And so on...

This is more heavyweight, and probably will be more suitable for developers who are keen to contribute more than just one PR. Hence, a good target audience would be Independent Work Module(IWM) and CS3281 students. However, it would also be useful for any external developers who wish to dive deeper. Thus I think this should be an optional (but highly recommended) material to go through for new devs.

One possible downside is the maintenance cost of keeping the onboarding tasks up-to-date, which we should keep in mind when drafting the materials.

Discussions/comments are welcomed:)

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by tlylt**
**What is the purpose of this pull request?**

- [x] Documentation update
- [ ] Bug fix
- [ ] Feature addition or enhancement
- [ ] Code maintenance
- [ ] DevOps
- [x] Improve developer experience
- [ ] Others, please explain:

**Overview of changes:**
Fixes #1997

- some adjustments to the structure of dev docs and necessary changes due to the shift
- add onboarding bootcamp

**Anything you'd like to highlight/discuss:**
This is my first time writing "bootcamp" materials, proofread, feedback and suggestions are welcomed.

**Testing instructions:**
https://deploy-preview-2063--markbind-master.netlify.app/devguide/bootcamp/intro

**Proposed commit message: (wrap lines at 72 characters)**
Create onboarding BootCamp

Currently, new developers may have difficulty ramping up and
contributing to the project due to a lack of guidance and
structure in the onboarding process.

In order to improve efficiency and effectiveness, new developers 
may want clear tasks and checkpoints to follow as they familiarize
themselves with the codebase and development workflow.

Let's create an onboarding bootcamp to guide new developers 
through the development workflow and codebase considerations.

This will help new developers ramp up quickly and contribute
effectively to the project.

---

**Checklist:** :ballot_box_with_check:

- [x] Updated the documentation for feature additions and enhancements
- [ ] Added tests for bug fixes or features
- [x] Linked all related issues
- [ ] No unrelated changes

### CODE REVIEW

**Comment by damithc** [comment]
Thanks for initiating this @tlylt

File: docs/devGuide/bootcamp/exploreMarkBind.md [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+
+## TLDR
+
+- [ ] Setup the master branch MarkBind
+- [ ] Create a MarkBind site
+- [ ] Create content with MarkBind
+- [ ] Modify site structure and configuration
+- [ ] Deploy your MarkBind site
+
+## Setup the master branch MarkBind
```
Comment (jonahtanjz) [inline docs/devGuide/bootcamp/exploreMarkBind.md:26]: Should this be the following instead?

```suggestion
## Setup the master branch of MarkBind
```

File: docs/devGuide/bootcamp/exploreMarkBind.md [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+
+<box type="important" light>
+
+**Checkpoint**
+
+After setting up MarkBind, try running from the root directory of your MarkBind repository:
+- [ ] `markbind -v` to check that you have the correct version of MarkBind installed
+- [ ] `cd docs && markbind serve -d` to check that you can serve the MarkBind documentation site in development mode
+
+If you can browse the MarkBind documentation site at http://localhost:8080, you have successfully set up MarkBind!
```
Comment (jonahtanjz) [inline docs/devGuide/bootcamp/exploreMarkBind.md:48]: Very minor nit: I believe if port 8080 is already in use (maybe some other server is using it on the user's computer), MarkBind will be served from another port. 

The URL address should be shown in the CLI output so perhaps we can mention this instead?

File: docs/devGuide/bootcamp/exploreMarkBind.md [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+1. Change `Your Logo` to `MarkBind Developer Test Site`.
+1. Change `Home :house:` to `My Test Playground :computer:`.
+
+<box type="important" light>
+
+**Checkpoint**
+
+Ensure that you are still serving the site with `markbind serve -d`. Check that the following are true:
+- [ ] The rendered page logo in the header is changed to `MarkBind Developer Test Site`
+- [ ] The rendered site navigation title is changed to `My Test Playground` :computer:
```
Comment (jonahtanjz) [inline docs/devGuide/bootcamp/exploreMarkBind.md:168]: How about adding some screenshots in the checkpoint area to help user better visualize what changes to look out for? 

Same for the above 2 checkpoints as well :)

File: docs/devGuide/bootcamp/contributeToDocs.md [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+You may want to check if there are any open issues tagged with the `a-Documentation` label. These [issues](https://github.com/MarkBind/markbind/issues?q=is%3Aopen+label%3A%22a-Documentation+%F0%9F%93%9D%22+sort%3Aupdated-desc) are usually related to documentation and may be a good place to start contributing for real!
+
+</box>
+
+<box type="important" light>
+
+**Checkpoint**
+
+Ensure that you can serve the MarkBind documentation site in development mode. You should be able to browse the site at http://localhost:8080.
```
Comment (jonahtanjz) [inline docs/devGuide/bootcamp/contributeToDocs.md:45]: Same comment as above.

File: docs/devGuide/bootcamp/contributeToDocs.md [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+Now, you can make a pull request targeting the `master` branch of the MarkBind repository.
+
+<div id="pr-steps">
+
+1. Fill in the pull request template.
+1. Wait for the CI checks to pass.
+   1. For first-time contributors, you may need to wait for the MarkBind team to approve your PR before the CI checks can run.
+1. Verify that the PR preview is working.
+   1. The PR preview is generated by netlify/markbind-master/deploy-preview.
+   1. You can find the link to the PR preview in the checks section of the PR.
```
Comment (jonahtanjz) [inline docs/devGuide/bootcamp/contributeToDocs.md:103]: I think a screenshot on where to find the PR preview link will be useful here :)

**Review by jonahtanjz — COMMENTED** [review]
Thanks @tlylt for coming up with this 👍 

It is really well done and clear. Good job 💯 

Generally looks good to me, just a couple of minor suggestions :)

**Comment by tlylt** [comment]
> Generally looks good to me, just a couple of minor suggestions :)

Thanks @jonahtanjz, I have applied the changes.

Was previously omitting the screenshots as I have referenced my example repository in the summary section and thought that it might be enough. I have since moved the mentioning of that example repo and the deployed site in front and added screenshots of the particular changes that are expected.

**Review by jonahtanjz — APPROVED** [review]
LGTM 👍

**Pull request merged**